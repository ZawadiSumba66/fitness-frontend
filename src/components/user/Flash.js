import { useState, useEffect } from 'react';
import Bus from '../../actions/bus';

const Flash = () => {
  const [visibility, setVisibility] = useState(false);
  const [message, setMessage] = useState('');
  const [type, setType] = useState('');

  useEffect(() => {
    Bus.addListener('flash', ({ message, type }) => {
      setVisibility(true);
      setMessage(message);
      setType(type);

      setTimeout(() => {
        setVisibility(false);
      }, 4000);
    });
  }, []);

  useEffect(() => {
    if (document.querySelector('.close') !== null) {
      document.querySelector('.close')
        .addEventListener('click', () => setVisibility(false));
    }
  });

  if (visibility) {
    return (
      <div className={`alert alert-${type} alert__wrapper w-100`}>
        <span className="close">
          <strong className="bg-dark">X</strong>
        </span>
        <p>{message}</p>
      </div>
    );
  }

  return <div />;
};

export default Flash;
