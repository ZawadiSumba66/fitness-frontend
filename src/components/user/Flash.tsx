import { useState, useEffect } from 'react';
import Bus from '../../actions/bus';
/* eslint-disable @typescript-eslint/no-shadow */
const Flash = () => {
  const [visibility, setVisibility] = useState(false);
  const [message, setMessage] = useState('');
  const [type, setType] = useState('');

  useEffect(() => {
    Bus.addListener('flash', ({ message, type }) => {
      setVisibility(true);
      setMessage(message);
      setType(type);

      setTimeout(() => {
        setVisibility(false);
      }, 3000);
    });
  }, []);

  if (visibility) {
    return (
      <div className={`alert alert-${type} alert__wrapper w-100`}>
        <p>{message}</p>
      </div>
    );
  }

  return <div />;
};

export default Flash;
/* eslint-enable @typescript-eslint/no-shadow */
