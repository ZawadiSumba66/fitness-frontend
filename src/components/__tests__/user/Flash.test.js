import renderer from 'react-test-renderer';
import Flash from '../../user/Flash';

it('matches the flash snapshot', () => {
  const tree = renderer.create(<Flash />).toJSON();
  expect(tree).toMatchSnapshot();
});
