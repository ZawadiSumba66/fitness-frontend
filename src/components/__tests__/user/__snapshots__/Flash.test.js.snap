// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the flash snapshot 1`] = `<div />`;
